metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/OEmbedProcessor.java

```diff
@@ -0,0 +1,232 @@
+   * @return the oembed model from json
+   * @throws IOException the io exception
+   */
+  public static OEmbedModel getOEmbedModelfromJson(byte[] jsonResource) throws IOException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    return objectMapper.readValue(jsonResource, OEmbedModel.class);
+  }
+
+  /**
+   * Gets oembed model from xml.
+   *
+   * @param xmlResource byte[]
+   * @return the oembed model from xml
+   * @throws IOException the io exception
+   */
+  public static OEmbedModel getOEmbedModelfromXml(byte[] xmlResource) throws IOException {
+    XmlMapper xmlMapper = new XmlMapper();
+    xmlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    return xmlMapper.readValue(xmlResource, OEmbedModel.class);
+  }
+
+  /**
+   * Is valid oembed photo or video boolean.
+   *
+   * @param oEmbedModel the oembed model
+   * @return the boolean true complies the minimum required fields for each type
+   */
+  public static boolean isValidOEmbedPhotoOrVideo(OEmbedModel oEmbedModel) {
+    return hasValidVersion(oEmbedModel) && hasValidType(oEmbedModel);
+  }
+
+  /**
+   * Has valid type boolean.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the boolean
+   */
+  private static boolean hasValidType(OEmbedModel oEmbedModel) {
+    return (isValidTypePhoto(oEmbedModel) || isValidTypeVideo(oEmbedModel));
+  }
+
+  /**
+   * Is valid type photo boolean.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the boolean
+   */
+  private static boolean isValidTypePhoto(OEmbedModel oEmbedModel) {
+    return oEmbedModel != null && oEmbedModel.getType() != null
+        && oEmbedModel.getType().equalsIgnoreCase("photo")
+        && oEmbedModel.getUrl()!=null && !oEmbedModel.getUrl().isEmpty()
+        && (oEmbedModel.getWidth() > 0 && oEmbedModel.getHeight() > 0);
+  }
+
+  /**
+   * Is valid type video boolean.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the boolean
+   */
+  private static boolean isValidTypeVideo(OEmbedModel oEmbedModel) {
+    return oEmbedModel != null && oEmbedModel.getType() != null
+        && oEmbedModel.getType().equalsIgnoreCase("video")
+        && oEmbedModel.getHtml()!=null && !oEmbedModel.getHtml().isEmpty()
+        && (oEmbedModel.getWidth() > 0 && oEmbedModel.getHeight() > 0);
+  }
+
+  /**
+   * Has valid version boolean.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the boolean
+   */
+  private static boolean hasValidVersion(OEmbedModel oEmbedModel) {
+    return oEmbedModel != null && oEmbedModel.getVersion() != null
+        && oEmbedModel.getVersion().startsWith("1.0");
+  }
+
+  /**
+   * Gets duration from model.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the duration from model
+   */
+  private static Double getDurationFromModel(OEmbedModel oEmbedModel) {
+    Double duration;
```

Comment:
Why not primitive? `Double.parseDouble` return primitive.

Reply:
I chose it that way because it is connected to the `VideoResourceMetadata` that one is using the Wrapper class. In this specific class I can use the primitive then.

Reply:
done.

Reply:
verified